library/test/src/test_result.rs

```diff
@@ -53,26 +59,32 @@ pub(crate) fn calc_result<'a>(
                 .map(|e| &**e)
                 .or_else(|| err.downcast_ref::<&'static str>().copied());
 
+            let should_panic_location = if desc.source_file != "" {
+                format!("should panic with message{}\n      ", at_location)
+            } else {
+                "".to_string()
+            };
+
             if maybe_panic_str.map(|e| e.contains(msg)).unwrap_or(false) {
                 TestResult::TrOk
             } else if let Some(panic_str) = maybe_panic_str {
                 TestResult::TrFailedMsg(format!(
-                    r#"panic did not contain expected string
+                    r#"{should_panic_location}panic did not contain expected string
       panic message: `{panic_str:?}`,
- expected substring: `{msg:?}`"#
+      expected substring: `{msg:?}`"#
```

Comment:
This is an unrelated change and should be split out to a different commit, maybe even a different PR.

Also, unsure if we should do it.